tools/bin/ecflow/processes/main/meteorological/get_metdata.py

```diff
@@ -127,68 +137,295 @@
+                       "[0:1:%s][0:1:%s]" % (num_lon, num_lat))
+    vs_lastyear_ds = xr.open_dataset(vs_url_lastyear)
+
+    vs_total_ds = xr.concat([vs_lastyear_ds, vs_thisyear_ds], 'day')
+
+    vs_total_ds.wind_speed.attrs = vs_attrs
+    vs_total_ds.lat.attrs = lat_attrs
+    vs_total_ds.lon.attrs = lon_attrs
+    vs_total_ds.day.attrs = day_attrs
+    vs_total_ds.attrs = globe_attrs
+    vs_total_ds.to_netcdf(os.path.join(met_loc, 'vs.nc'),
+                          mode='w', format='NETCDF4')
+
+    # shortwave radiation
+    srad_url_thisyear = ("http://thredds.northwestknowledge.net:8080" +
+                         "/thredds/dodsC/MET/srad/srad_%s.nc?lon[0:1:%s]," % (year, num_lon) +
+                         "lat[0:1:%s],day[%s:1:%s]," % (num_lat, num_startofyear, num_day) +
+                         "surface_downwelling_shortwave_flux_in_air[%s:1:%s]" % (num_startofyear, num_day) +
+                         "[0:1:%s][0:1:%s]" % (num_lon, num_lat))
+    srad_thisyear_ds = xr.open_dataset(srad_url_thisyear)
+
+    srad_url_lastyear = ("http://thredds.northwestknowledge.net:8080" +
+                         "/thredds/dodsC/MET/srad/srad_%s.nc?lon[0:1:%s]," % (lastyear, num_lon) +
+                         "lat[0:1:%s],day[%s:1:%s]," % (num_lat, lastyear_num_day, num_endofyear) +
+                         "surface_downwelling_shortwave_flux_in_air[%s:1:%s]" % (lastyear_num_day, num_endofyear) +
+                         "[0:1:%s][0:1:%s]" % (num_lon, num_lat))
+    srad_lastyear_ds = xr.open_dataset(srad_url_lastyear)
+
+    srad_total_ds = xr.concat([srad_lastyear_ds, srad_thisyear_ds], 'day')
+
+    srad_total_ds.surface_downwelling_shortwave_flux_in_air.attrs = srad_attrs
+    srad_total_ds.lat.attrs = lat_attrs
+    srad_total_ds.lon.attrs = lon_attrs
+    srad_total_ds.day.attrs = day_attrs
+    srad_total_ds.attrs = globe_attrs
+    srad_total_ds.to_netcdf(os.path.join(met_loc, 'srad.nc'),
+                            mode='w', format='NETCDF4')
+
+    # specific humidity
+    sph_url_thisyear = ("http://thredds.northwestknowledge.net:8080" +
+                        "/thredds/dodsC/MET/sph/sph_%s.nc?lon[0:1:%s]," % (year, num_lon) +
+                        "lat[0:1:%s],day[%s:1:%s]," % (num_lat, num_startofyear, num_day) +
+                        "specific_humidity[%s:1:%s]" % (num_startofyear, num_day) +
+                        "[0:1:%s][0:1:%s]" % (num_lon, num_lat))
+    sph_thisyear_ds = xr.open_dataset(sph_url_thisyear)
+
+    sph_url_lastyear = ("http://thredds.northwestknowledge.net:8080" +
+                        "/thredds/dodsC/MET/sph/sph_%s.nc?lon[0:1:%s]," % (lastyear, num_lon) +
+                        "lat[0:1:%s],day[%s:1:%s]," % (num_lat, lastyear_num_day, num_endofyear) +
+                        "specific_humidity[%s:1:%s]" % (lastyear_num_day, num_endofyear) +
+                        "[0:1:%s][0:1:%s]" % (num_lon, num_lat))
```

Comment:
since you've just repeated nearly the same lines 12 times, I think it would make since to

1. put this into a helper function (e.g. `fill_threads_forcing_url(varname, *args)`)
2. read/fill/write these files within a for loop.